sdks/python/apache_beam/runners/worker/operations.py

```diff
@@ -600,7 +627,8 @@ def _read_side_inputs(self, tags_and_types):
           self.name_context.step_name,
           view_options['coder'],
           i,
-          suffix='side-input')
+          suffix='side-input',
+          producer_type_hints=get_perf_runtime_type_hints(self))
```

Comment:
This isn't correct, these type hints should not be used for this counter. Make this an optional argument defaulting to None and don't pass it here.

Reply:
Sounds good